### PR TITLE
Fixes #8329: Added support of options for `message` console command

### LINKED ISSUE #8329: Enh: extract messages for specified language
Because translator work mostly with one language, it should be better have ability extract messages for specified language easier than generate new config.php and then edit it.

After this patch translator can use default config.php "from the box" like below:
```
./yii message/extract messages/config.php langCode
```

Patch code:
```php
--- a/framework/console/controllers/MessageController.php
+++ b/framework/console/controllers/MessageController.php
@@ -76,9 +76,12 @@ class MessageController extends Controller
      * @param string $configFile the path or alias of the configuration file.
      * You may use the "yii message/config" command to generate
      * this file and then customize it for your needs.
+     * @param string $specifiedLanguage the language code.
+     * This parameter override 'languages' array content
+     * from $configFile with one specified language code.
      * @throws Exception on failure.
      */
-    public function actionExtract($configFile)
+    public function actionExtract($configFile, $specifiedLanguage = NULL)
     {
         $configFile = Yii::getAlias($configFile);
         if (!is_file($configFile)) {
@@ -93,6 +96,9 @@ class MessageController extends Controller
             'format' => 'php',
             'ignoreCategories' => [],
         ], require($configFile));
+        if (!empty($specifiedLanguage)) {
+            $config['languages'] = [ $specifiedLanguage ];
+        }
 
         if (!isset($config['sourcePath'], $config['languages'])) {
             throw new Exception('The configuration file must specify "sourcePath" and "languages".');
```

### PR DESCRIPTION
**Pull request by samdark**
Replaces https://github.com/yiisoft/yii2/pull/8739

### CODE REVIEW

**Comment by samdark** [comment]
@yiisoft/core-developers need review. Especially:

- `framework/console/Controller.php`.
- `framework/console/controllers/MessageController.php`.

File: framework/console/controllers/MessageController.php [hunk 1]
```diff
@@ -78,22 +245,25 @@ public function actionConfig($filePath)
      * this file and then customize it for your needs.
      * @throws Exception on failure.
      */
-    public function actionExtract($configFile)
+    public function actionExtract($configFile = null)
```
Comment (SilverFire) [inline framework/console/controllers/MessageController.php]: Signature changed. Should be mentioned in UPGRADE.md

Comment (SilverFire) [inline framework/console/controllers/MessageController.php]: Also we need to update to the PHPDoc and describe behavior in case when `$configFile` is null

**Comment by SilverFire** [comment]
Looks great.
I'm not sure whether we need to add `@since` everywhere :)

File: framework/console/controllers/MessageController.php (hunk 1 above)
Comment (samdark) [inline framework/console/controllers/MessageController.php]: No need to do it since console output would tell you what's wrong in this case.

**Pull request merged**